### ISSUE TITLE
Transformer configuration should be passed to the tokenizer? 

### ISSUE THREAD

**Issue by lfoppiano** [kermitt2/delft#144]
I'm noticing that some of the models I've been training, which are supposed to be `cased`, result in a delft model where `do_lower_case=True`. It seem that by default `do_lower_case` is set to True. 
Initially I though it was a bug in my code, as commented in #143, however it seems that such parameters are not propagated correctly to the output model. 

I'm not sure, but perhaps we should pass the transformer configuration also to the tokenizer, as I've done in [8ef0dff](https://github.com/kermitt2/delft/pull/143/commits/8ef0dffd6a57d9f5195cee870ac6905a226c0a4f)

**Comment by kermitt2** [comment]
Thanks, this is very annoying :( 

I looked at the saved fine-tuned SciBERT model loaded from Huggingface (`allenai/scibert_scivocab_cased`), and in the saved transformer tokenizer (`tokenizer_config.json`) we indeed have `do_lower_case: True` and in `tokenizer.json` we have `lower_case: true`. 

So, more generally, I understand that loading the tokenizer from the name only with huggingface library do not set correctly the case normalization (saved transformer vocab file have cased tokens). 

It would not just be for the local loading of a pretrained model on file. 

I vaguely remember to have checked the actual transformer tokens resulting from the application of the tokenizer when training, and I don't remember a problem (tokens were cased), but it is worth double checking :(

**Comment by lfoppiano** [comment]
In the huggingface models configuration, such information is not provided, however the model is clearly stated as "cased". So I wonder why is that? 
I've tested quickly and it seems no cased but I will check again tomorrow. 

I've asked there, maybe they can clarify... https://huggingface.co/allenai/scibert_scivocab_cased/discussions/2

**Comment by kermitt2** [comment]
Looking again at `allenai/scibert_scivocab_cased`, I checked the encoded/decoded tokens from the AutoTokenizer, and the tokens are indeed all uncased, consistent with the config files of the tokenizer coming with model indicates `do_lower_case: True` (`tokenizer_config.json`) and `lower_case: true` (`tokenizer.json`).

```
input: ['Puterman', ',', 'M', '.', 'L', '.', '(', '1994', ')', '.', 'Markov', 'Decision', 'Problems', '.', 'Wiley', ',', 'New', 'York', '.']
id_to_tokens: ['[CLS]', 'put', '##erman', ',', 'm', '.', 'l', '.', '(', '1994', ')', '.', 'mark', '##ov', 'decision', 'problems', '.', 'wi', '##ley', ',', 'new', 'y', '##ork', '.', '[SEP]']
decode: [CLS] puterman, m. l. ( 1994 ). markov decision problems. wiley, new york. [SEP]
```

I checked the same for `bert-base-cased`, we have `"do_lower_case": false` in `tokenizer_config.json` as expected and lower_case: false` in `tokenizer.json`.

So the Tokenizer config of the scibert model saved on HuggingFace is indeed incorrect, it an uncased tokenizer, while the vocabulary is cased and the model is supposed to work with the cased token ids.

**Comment by lfoppiano** [comment]
Thanks @kermitt2. Indeed there is a problem with the scibert model in HuggingFace... Adding the tokenizer_config.json (borrowed from the bert-cased) the configuration is correctly setting the cased parameters. 

I try to open a PR with the missing file to huggingface, but eventually I have to re-evaluate all the models again 😭 since I guess the casing might have some impact in sequence labelling.
